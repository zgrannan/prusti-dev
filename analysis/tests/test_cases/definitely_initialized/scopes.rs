#[analyzer::run]
fn main() {
    {
        let x = 123;
    }
    {
        let y = 456;
    }
}
