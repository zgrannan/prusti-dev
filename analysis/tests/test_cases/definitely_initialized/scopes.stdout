Analyzing file $DIR/scopes.rs using DefinitelyInitializedAnalysis...
Result for function main():
{
  "bb0": [
    [
      [
        "state:",
        [],
        "statement: StorageLive(_1)"
      ],
      [
        "state:",
        [],
        "statement: StorageLive(_2)"
      ],
      [
        "state:",
        [],
        "statement: _2 = const 123_i32"
      ],
      [
        "state:",
        [
          "_2"
        ],
        "statement: FakeRead(ForLet(None), _2)"
      ],
      [
        "state:",
        [
          "_2"
        ],
        "statement: _1 = const ()"
      ],
      [
        "state:",
        [
          "_1",
          "_2"
        ],
        "statement: StorageDead(_2)"
      ],
      [
        "state:",
        [
          "_1"
        ],
        "statement: StorageDead(_1)"
      ],
      [
        "state:",
        [],
        "statement: StorageLive(_3)"
      ],
      [
        "state:",
        [],
        "statement: _3 = const 456_i32"
      ],
      [
        "state:",
        [
          "_3"
        ],
        "statement: FakeRead(ForLet(None), _3)"
      ],
      [
        "state:",
        [
          "_3"
        ],
        "statement: _0 = const ()"
      ],
      [
        "state:",
        [
          "_0",
          "_3"
        ],
        "statement: StorageDead(_3)"
      ]
    ],
    "state before terminator:",
    [
      "_0"
    ],
    "terminator: return",
    {}
  ]
}
